### DISCUSSION TITLE
Retreiving data chartd/datasets/dashboards based on

### DISCUSSION THREAD

**Discussion by supermarin1** [apache/superset#18276] [Q&A / Help]
Is there the way to export data charts/dashboards based on with REST API v. 1.0.0?

Tried /chart/export/, /database/export/, /dashboard/export/ and have no luck. Or I understand wrong meaning of export?

Please, advise.

**Comment by junlincc** [comment]
@dpgaspar ^ Daniel could you help answer? thanks!

**Comment by willbarrett** [comment]
@betodealmeida would you mind weighing in?

**Comment by supermarin1** [comment]
please, anybody?

**Comment by GGPay** [comment]
I've used **dashboard** with q parameters  `{
  "order_column": "changed_on_delta_humanized",
  "order_direction": "desc",
  "page": 0,
  "page_size": 25
}`

Got response 
![image](https://user-images.githubusercontent.com/17413180/111582648-2078f780-8789-11eb-8455-9746d1188e03.png)


And then make a request - where id in parentheses is ID of dashboard

http://127.0.0.1:8088/api/v1/dashboard/export/?q=!(3)

Wouldn't say it's proper solution but that the way how i made it work.

**Comment by supermarin1** [comment]
@GGPay thanks

I can get all info about dashboard/chart/dataset but I'm interesting in possibility to get/export data dashboard/chart/dataset based on. 

I tried all API calls and did not find any way to retrieve data at least from DB. Is this possible?

**Comment by GGPay** [comment]
As i said before - open url in a browser where 142 - is chart ID

Don't forget authenticate.

http://your_superset_host/api/v1/chart/export/?q=!(142)

3 - Dashboard ID

http://your_superset_host/api/v1/dashboard/export/?q=!(3)

And get yaml zip file

**Comment by supermarin1** [comment]
I've got .zip file with .yaml files which are schemas, not values.

May be smth need to be configured in additional way?

**Comment by GGPay** [comment]
I don't really understand what are you trying to achieve? Dashboard/Charts/Queries export are export of .yamls schemas. Don't understand why you need export values - just write SQL to database.

Take a look chart/data  or reverse frontend request to find what you're looking.

**Comment by supermarin1** [comment]
And I can't get data from DB through the API?

**Comment by dpgaspar** [comment]
@supermarin1 like @GGPay take a looks at `api/v1/chart/data`

**Comment by supermarin1** [comment]
Yes! That what I was looking for. Thanks @GGPay @dpgaspar

**Comment by supermarin1** [comment]
The next question:

I have schema and information for ordering from `/api/v1/chart/export?q=[chat_ds]` (please note order columns is empty)
`  order_by_cols: [],
  order_desc: true`
  
  Forming the json for `api/v1/chart/data` POST request ` "order_desc": true ` doesn't provide any ordering. Changing to `false` doesn't make any impact. 
  
  How can I get the proper ordering?

**Comment by GGPay** [comment]
I'm not really famililar with `api/v1/chart/data` endpoint but have look tests - it's really useful

`superset/tests/charts/api_tests.py`

**Comment by supermarin1** [comment]
Thera are no test case covering my case :-( And this is more about `/api/v1/chart/export` endpoint. 

I'm able to get all info from `api/v1/chart/export?q=[chart_id]` to form post request to `/api/v1/chart/data` **except** ordering columns.

There is no chart settings as for sorting
![image](https://user-images.githubusercontent.com/24755510/112824371-1f9e6a80-9093-11eb-9024-050579a58d90.png)
but there is sorting in SQL query (chart -> view query)
![image](https://user-images.githubusercontent.com/24755510/112824530-507e9f80-9093-11eb-9a09-9a8edab1672d.png)

As the result - I can't get the same result with API as I can see on UI.

May be there is the way we can get that query?
